sound/soc/sof/topology.c

```diff
@@ -936,6 +953,11 @@ static int sof_parse_tokens(struct snd_soc_component *scomp,
 		/* next array */
 		array = (struct snd_soc_tplg_vendor_array *)((u8 *)array
 			+ asize);
+		pos += asize;
+
+		/* return position where the array was found */
+		if (found == count)
+			return pos;
```

Comment:
doesnt' look like any of the routines below actually use the position for anything, they just test the status. Why is this needed? or what I am missing?

Reply:
the mux/demux parsing uses it. nobody else doesn't currently.